pkg/api/kubernetesagentpool/validate.go

```diff
@@ -0,0 +1,99 @@
+package kubernetesagentpool
+
+import (
+	"fmt"
+	"strings"
+)
+
+func (p *AgentPool) Validate() error {
```

Comment:
I wonder if these validations would be more helpful to the user if the exact string that would need to be set in json would be contained in the error message? (Or does it already and I just didn't know some magic?)